storage-proofs/core/src/settings.rs

```diff
@@ -34,6 +35,7 @@ impl Default for Settings {
     fn default() -> Self {
         Settings {
             maximize_caching: true,
+            verify_cache: false,
```

Comment:
It may cost miners hours of time per sector, so instead of default to true, it should be advised to check when the error cases arise.  My fastest machine can verify all sizes in 2.5 hours, which is more than is needed (if enabled), but almost all of the time is spent on the 32&64GiB verifications, so it'll add up quickly.

Reply:
Got it. Let's make sure to document clearly and with sufficient visibility that those will benefit from this know when they need to enable.

Also, related to my suggestion elsewhere, I think we can hide all or most of this cost by generating the digest incrementally (in parallel with graph computation) while generating the cache?

If we can do that, we could afford to *always* check integrity on verification (best!). That would make the supplementary check here only truly relevant in the case of disk corruption, and a better candidate for being enabled only when debugging.

Reply:
The more I think about it, the more I think this optimization is important for maximizing usability of this integrity check. If it's intrusively expensive, people won't enable it by default, and we'll miss the assurance it provides. Unless it proves impossible to keep this fast (as above), I think the integrity check at generation time should be mandatory.

Reply:
I like the idea, but I'm not sure how to do it given how the cache is created in parallel at the moment.  Generation on my fastest machine takes closer to 24+ hours IIRC (and closer to 40 or so on my normal dev machine), I was referring to the verification only in the timing above (i.e. assuming you have files on disk, how long to verify they are good).

Reply:
If cache-generation is already parallelized, then this won't help, but maybe with release mode it's not such an issue.

```
➜  filecoin-parents /usr/bin/time sha256sum v28-sdr-parent-d5500bc0dddadb609f867d94da1471ecbaac3fe6f8ac68a4705cebde04a765b8.cache
f6deb8b7804ad7327ab9eb86b6b66f7dbb7dfc370cac54e1b5267984105d3611  v28-sdr-parent-d5500bc0dddadb609f867d94da1471ecbaac3fe6f8ac68a4705cebde04a765b8.cache
27.56user 7.10system 0:34.73elapsed 99%CPU (0avgtext+0avgdata 3140maxresident)k
5272inputs+0outputs (1major+157minor)pagefaults 0swaps
➜  filecoin-parents ll v28-sdr-parent-d5500bc0dddadb609f867d94da1471ecbaac3fe6f8ac68a4705cebde04a765b8.cache
-rw-r--r-- 1 xxx xxx 56G Aug 28 00:36 v28-sdr-parent-d5500bc0dddadb609f867d94da1471ecbaac3fe6f8ac68a4705cebde04a765b8.cache
```
The above assumes a machine with fast SHA256, but I think that holds for miners who will be generating the parents cache anyway. ~30 seconds is not so bad.

Reply:
I just ran the verify on my mining rig for 32GiB (in release mode) and it took just under 32 seconds, so we seem to agree there.